orderer/common/multichannel/registrar.go

```diff
@@ -581,11 +581,42 @@ func (r *Registrar) joinAsFollower(ledgerRes *ledgerResources, clusterConsenter
 	return info, nil
 }
 
+type externalHalter interface {
+	HaltExternal()
+}
+
 // RemoveChannel instructs the orderer to remove a channel.
 // Depending on the removeStorage parameter, the storage resources are either removed or archived.
 func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
 	if r.SystemChannelID() != "" && channelID != r.SystemChannelID() {
 		return types.ErrSystemChannelExists
 	}
-	return errors.New("Not implemented yet")
+
+	cs, ok := r.chains[channelID]
+	if !ok {
+		return types.ErrChannelNotExist
+	}
```

Comment:
a follower may also be removed, not just a chain.

Reply:
Done. Haven't added the tests yet (just noticed your join channel as follower PR was merged so I'll use that as a reference).

Reply:
Added the UT for this (a while ago).